### PR TITLE
C++: Demonstrate overflowing guard bounds

### PR DESCRIPTION
**Pull request by jbj**
I identified this problem while working on #4420. I'm not sure if it's ever a problem in practice, but I thought I'd record it as a test case.

### CODE REVIEW

File: cpp/ql/test/library-tests/rangeanalysis/SimpleRangeAnalysis/test.c [hunk 1]
```diff
@@ -638,4 +638,16 @@ void two_bounds_from_one_test(short ss, unsigned short us) {
...
     out(ss); // -1 .. 2
   }
-}
\ No newline at end of file
+}
+
+void guard_bound_out_of_range(void) {
+  int i = 0;
+  if (i < 0) {
+    out(i); // 0 .. 0 [BUG: is -max .. +max]
```
Comment (geoffw0) [inline cpp/ql/test/library-tests/rangeanalysis/SimpleRangeAnalysis/test.c:646]: This is unreachable, should there be any bounds?

Comment (jbj) [reply cpp/ql/test/library-tests/rangeanalysis/SimpleRangeAnalysis/test.c]: You're right. Fixed.

**Review by geoffw0 — APPROVED** [review]
:+1:

**Comment by geoffw0** [comment]
I was about to merge this but there are merge conflicts.

**Comment by jbj** [comment]
Yes, it conflicted with #4420. Resolved.

**Review by geoffw0 — APPROVED** [review]
:+1:

**Pull request merged**